### PR TITLE
register, cluster iterators

### PR DESCRIPTION
**Pull request by burrbull**

### CODE REVIEW

File: svd-rs/CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,8 @@ and this project adheres to [Semantic Versioning](http://semver.org/).
 
 ## Unreleased
 
+- Rename `reg_iter` to `all_registers`, add `registers`, `clusters` methods that create itertors
```
Comment (Emilgardis) [inline svd-rs/CHANGELOG.md:10]: Typo, `iterators`

File: svd-rs/src/clusterinfo.rs [hunk 2]
```diff
@@ -226,21 +229,61 @@ impl ClusterInfo {
         Ok(())
     }
 
-    /// returns a iterator over all registers the cluster contains
-    pub fn reg_iter(&self) -> RegIter {
+    /// Returns iterator over all descendant registers
+    #[deprecated(note = "Please use `all_registers` instead")]
```
Comment (Emilgardis) [inline svd-rs/src/clusterinfo.rs:233]: deprecated attr should have a since field

**Comment by burrbull** [comment]
bors retry

**Review by Emilgardis — APPROVED** [review]
bors r+

**Comment by burrbull** [comment]
bors retry

**Pull request merged**